tensorflow/tensorflow.bzl

```diff
@@ -379,13 +385,43 @@ def _rpath_linkopts(name):
 
 # Bazel-generated shared objects which must be linked into TensorFlow binaries
 # to define symbols from //tensorflow/core:framework and //tensorflow/core:lib.
-def tf_binary_additional_srcs():
-    return if_static(
-        extra_deps = [],
-        otherwise = [
-            clean_dep("//tensorflow:libtensorflow_framework.so"),
+def tf_binary_additional_srcs(fullversion = False):
+    if fullversion:
+        suffix = "." + VERSION
+    else:
+        suffix = "." + VERSION.split(".")[0]
+
+    return select({
+        clean_dep("//tensorflow:macos_with_framework_shared_object"): [
+            clean_dep("//tensorflow:libtensorflow_framework%s.dylib" % suffix),
         ],
-    )
+        clean_dep("//tensorflow:framework_shared_object"): [
+            clean_dep("//tensorflow:libtensorflow_framework.so%s" % suffix),
+        ],
+        "//conditions:default": [
+        ],
+    })
+
+# Helper function for the per-OS tensorflow libraries and their version symlinks
+def tf_shared_library_deps():
+    longsuffix = "." + VERSION
+    suffix = "." + VERSION.split(".")[0]
+
+    return select({
+        clean_dep("//tensorflow:macos"): [
+            clean_dep("//tensorflow:libtensorflow.dylib"),
+            clean_dep("//tensorflow:libtensorflow%s.dylib" % suffix),
+            clean_dep("//tensorflow:libtensorflow%s.dylib" % longsuffix),
+        ],
+        clean_dep("//tensorflow:windows"): [
+            clean_dep("//tensorflow:tensorflow.dll"),
+        ],
+        "//conditions:default": [
+            clean_dep("//tensorflow:libtensorflow.so"),
```

Comment:
I am surprised we need this. Does genrule not automatically create dependencies to its "srcs"?
I think taking away the ability to create a dependence on libtensorflow.so can be quite confusing if anyone is using it through bazel.

Reply:
NVM, shared library dependencies do not really get propagated by bazel. Now I remember...

Reply:
yeah they dont. I've been talking to the bazel guys tho a bit so we'll make it nicer in the future but for now we need this.